src/plugins/stake-plugins/generic/policyAdapters/EthereumKilnAdaptor.ts

```diff
@@ -74,7 +74,41 @@ export const makeKilnAdapter = (policyConfig: StakePolicyConfig<EthereumPooledKi
     stakePolicyId,
 
     async fetchClaimQuote(wallet: EdgeCurrencyWallet, requestAssetId: AssetId, nativeAmount: string): Promise<ChangeQuote> {
-      throw new Error('fetchClaimQuote not implemented')
+      const { maxFeePerGas, maxPriorityFeePerGas, nextNonce, walletSigner, walletAddress } = await workflowUtils(wallet)
+      const { currencyCode, pluginId } = wallet.currencyInfo
+
+      const allocations: QuoteAllocation[] = []
+
+      const operations = await kiln.getOperations(baseUrl, walletAddress)
+
+      let claimableTotal = '0'
+      const ticketIds: string[] = []
+      const caskIds: string[] = []
+      for (const operation of operations) {
+        if (operation.type === 'exit' && operation.ticket_status === 'fulfillable') {
+          claimableTotal = add(claimableTotal, operation.size)
+          ticketIds.push(operation.ticket_id)
+          // use lowest uint32 per ticket https://docs.kiln.fi/v1/kiln-products/on-chain/pooled-staking/how-to-integrate/staking-interactions/unstaking-and-withdrawals#claim-tickets-once-their-fulfillable
+          const lowestCaskId = operation.cask_ids.reduce((a, b) => (lt(a, b) ? a : b))
+          caskIds.push(lowestCaskId)
+        }
+      }
+
+      allocations.push({
+        allocationType: 'claim',
+        pluginId,
+        currencyCode,
+        nativeAmount: claimableTotal
+      })
+
+      const tx = await integrationContract.populateTransaction.multiClaim([exitQueueAddress], [ticketIds], [caskIds], {
+        gasLimit: '500000',
```

Comment:
Why do these additions in this commit have static gasLimits but `.stake` contract method doesn't?

Reply:
Nodes return an error when estimating the gas limit for these transactions. They seem to be able to estimate the stake transaction without issue.